cloud/blockstore/libs/daemon/ydb/bootstrap.cpp

```diff
@@ -433,23 +434,30 @@ void TBootstrapYdb::InitKikimrService()
 
     InitSpdk();
 
-    FileIOService = CreateAIOService();
-
-    if (Configs->DiskAgentConfig->GetEnabled() &&
-        Configs->DiskAgentConfig->GetBackend() == NProto::DISK_AGENT_BACKEND_AIO &&
+    if (const auto& config = *Configs->DiskAgentConfig;
+        config.GetEnabled() &&
+        config.GetBackend() == NProto::DISK_AGENT_BACKEND_AIO &&
         !AioStorageProvider)
     {
-        Y_ABORT_UNLESS(FileIOService);
-
         NvmeManager = CreateNvmeManager(
             Configs->DiskAgentConfig->GetSecureEraseTimeout());
 
+        auto factory = [events = config.GetMaxAIOContextEvents()] {
+            return CreateAIOService(events);
+        };
+
+        FileIOServiceProvider =
+            config.GetFilePathsPerIOServiceCount()
+                ? CreateFileIOServiceProvider(
+                      config.GetFilePathsPerIOServiceCount(),
+                      factory)
+                : CreateFileIOServiceProviderStub(factory());
```

Comment:
It's not so clear that stub functionality is to provide same file io for all paths so maybe rename it to `CreateSingleFileIOServiceProvider` or even call `CreateFileIOServiceProvider` with `std::numeric_limits<uint32_t>::max()` but watch the overflow in computations